### PR TITLE
fix: Remove percent sign from IDs

### PR DESCRIPTION
**Pull request by nschonni**
Related to https://github.com/heycam/webidl/pull/955

### CODE REVIEW

**Comment by devsnek** [comment]
If these are changed, they need `oldid` entries (`<x oldids="oldid1,oldid2,etc">`). That being said, I don't think they should be changed, they're totally valid ids.

**Comment by bakkot** [comment]
> they need `oldid` entries

And doing so will cause elements with `%` to be generated anyway, so there's no way to eliminate them without breaking links.

---

In any case, what's wrong with `%`? The HTML spec [says](https://html.spec.whatwg.org/multipage/dom.html#global-attributes)

> the id attribute value must be unique amongst all the IDs in the element's tree and must contain at least one character. The value must not contain any ASCII whitespace. [...] There are no other restrictions on what form an ID can take; in particular, IDs can consist of just digits, start with a digit, start with an underscore, consist of just punctuation, etc.

**Comment by nschonni** [comment]
You're right that it isn't a restricted character anymore in HTML5, but `%` has meaning in the end URL which these are used as the anchors.
Realized I also didn't hit all the instances with a quick find/replace regex as well

File: spec.html [hunk 1]
```diff
@@ -33139,7 +33139,7 @@ <h1>The %TypedArray% Intrinsic Object</h1>
         <li>will throw an error when invoked, because it is an abstract class constructor. The _TypedArray_ constructors do not perform a `super` call to it.</li>
       </ul>
 
-      <emu-clause id="sec-%typedarray%">
+      <emu-clause id="sec-typedarray">
```
Comment (jmdyck) [inline spec.html:33142]: Collides with existing id.

File: spec.html [hunk 2]
```diff
@@ -33219,7 +33219,7 @@ <h1>%TypedArray%.of ( ..._items_ )</h1>
         </emu-alg>
       </emu-clause>
 
-      <emu-clause id="sec-%typedarray%.prototype">
+      <emu-clause id="sec-typedarray.prototype">
```
Comment (jmdyck) [inline spec.html:33222]: Collides with existing id.

File: spec.html [hunk 3]
```diff
@@ -33288,12 +33288,12 @@ <h1>get %TypedArray%.prototype.byteOffset</h1>
         </emu-alg>
       </emu-clause>
 
-      <emu-clause id="sec-%typedarray%.prototype.constructor">
+      <emu-clause id="sec-typedarray.prototype.constructor">
```
Comment (jmdyck) [inline spec.html:33291]: Collides with existing id.

**Review by jmdyck — CHANGES_REQUESTED** [review]

**Comment by ljharb** [comment]
@nschonni what meaning does it have?

**Comment by nschonni** [comment]
The beginning of a percent encoded bytes https://url.spec.whatwg.org/#percent-encoded-bytes
Although I think none of the find/replaced ones sequential HEX characters

**Comment by ljharb** [comment]
@nschonni has that always been the case? or did adding that feature break the web, at least the part of the web that's using %'s in anchors, like this part?

**Comment by bakkot** [comment]
@ljharb Percent-encoding in URLs has been a thing for a very long time.

**Comment by ljharb** [comment]
ah, true. but the URLs that we use, with the `%`, have worked for a long time as well. What's the impetus to change them now?

**Comment by devsnek** [comment]
> And doing so will cause elements with % to be generated anyway; there's no way to eliminate %s in ids in the generated document without breaking links.

@bakkot my understanding is that this PR is here to enable the links in webidl to pass some sort of html validator, not emca262 itself (which one is yet unclear, like as you and i said above, the ids are valid)

**Comment by mathiasbynens** [comment]
Based on the description in https://github.com/heycam/webidl/pull/955, @nschonni is trying to address an *upstream* HTML validation issue. I’m guessing it’s the following:

```html
<a href="https://tc39.es/ecma262/#sec-%typedarray%-intrinsic-object">foo</a>
```

> Error: Bad value `https://tc39.es/ecma262/#sec-%typedarray%-intrinsic-object` for attribute `href` on element `a`: Percentage ("%") is not followed by two hexadecimal digits.

That said, if upstream HTML validation is the goal, then rather than making changes to ECMA-262, the upstream HTML document should be patched (e.g. by URL-encoding the `%` sign as `%25` in the above HTML snippet).

**Comment by bakkot** [comment]
Closing in light of @mathiasbynens's comment.

**Pull request closed without merging**